### ISSUE TITLE
Make use of callbacks in the Neural Network Classifier & Regressor tutorial

### ISSUE THREAD

**Issue by adekusar-drl** [Qiskit/qiskit-machine-learning#195]
### What is the expected enhancement?
Training a classifier/regressor in the "Neural Network Classifier & Regressor" tutorial may take some time, so to make the tutorial more interactive and provide some insights into the training process I suggest to make use of the recently introduced callback interface and collect objective(loss) function values obtained from an optimizer. These values can be just printed, thus making the notebook interactive and/or plotted. Thus, we will achieve two goals: dynamic behaviour of the tutorial during long training processes and show how to use callbacks.
